stable/jenkins/templates/config.yaml

```diff
@@ -36,7 +36,13 @@ data:
               <instanceCap>2147483647</instanceCap>
               <idleMinutes>0</idleMinutes>
               <label>{{ .Release.Name }}-{{ .Values.Agent.Component }}</label>
-              <nodeSelector></nodeSelector>
+              <nodeSelector>
+                {{- $local := dict "first" true }}
```

Comment:
Yeah, gotl is horrible at this sort of thing. this is the only way I've figured out how to do this.... 

Gotl out of the box has no way of determining if you are in the first iteration (or last) iteration of a loop or not. But you need to know if you need to put comma's or some other thing on everything not the first run(or visa versa)

gotl's scoping rules also do not facilitate this sort of thing. You can't change the value of something already set. so you can't just 
{{ $first := true}}
range ....
{{ $first := false }}
end range

But, there is one caviot. Using the additional 'set' function, you can change the value of a key stored in a dict. This lets the $local.first value be overwritten inside the loop letting it keep state of whether its been through the loop once already or not, thus allowing commas to be added in the right places.

Yeah. its kind of a hack, but it works. haven't found a way to do it better.